### ISSUE TITLE
FileLumaToAlpaha filter affects audio levels.

### ISSUE THREAD

**Issue by doradoro** [jliljebl/flowblade#766]
Please include the following information.
1) Flowblade version (Help->About): 2.2.0
2) MLT version (Help->Runtime Environment): 6.14.0
3) Your distribution (Ubuntu, Debian, Mint etc.): Manjaro
4) If possible, try to provide step-by-step instructions on how to reproduce the issue. From developer point of view this is the easieast way to understand what is happening.:

- I have a wav file created with audacity.
- I play it outside flowblade and sounds at full volume
- i open my project, add it and the sound volume drops
- i play it in the editor tab and the volume changes weirdly
- i create another project, add it and i get the same result
- i render the file, open it in audacity and the volume wave is about 25% of its "capacity"
- i do it in Olive and sound normally (it also uses mlt)

I dont know what to do and i dont know how it happened. i deleted the flowflade .config dir = same result. i checked the mixer values = all 100%. I reinstalled mlt, same result. I reinstalled flowblade, same result! i tried flatpak version, same result! i tried different files and formats, same result!

I need to upload my video today... :(

**Comment by jliljebl** [comment]
Could you make the problem file available for download somehow?

**Comment by doradoro** [comment]
This time only got a bit lower but in my actual project is like four times lower

https://gofile.io/?c=j0Jzk1

**Comment by jliljebl** [comment]
This simply does not reproduce here. 

All files sound exactly same when played back from Audacity and Flowblade, rendered clips from audio files you made available sound exactly the same as  the original clips when placed on timeline after them. Nothing like you described is happening.

Overall I'm getting an absolute s***ton of reports from Manjaro systems that do not reproduce, I'm 90% sure that the issue here is with Manjaro, not Flowblade.

I'm leaving this open as *unreprodcued* but will close in a few weeks unless I'm getting some other evidence that this really is a Flowblade issue.

**Comment by jliljebl** [comment]
I'm not getting any of terminal printouts that you have either.

**Comment by doradoro** [comment]
I willl try to reinstall pulse and those things...

**Comment by dvdlvr** [comment]
I can't hear a marked difference between the sound files on my system (Ubuntu).
Though I noticed that test.wav is only mono whereas the other two wav files are stereo.
I re-rendered the mkv file and heard no loss in the audio level.

**Comment by doradoro** [comment]
Its only a bit in that example but in my actual project sounds about the 25% of total volume... I think that something is lowering the 50% each time two layers are mixed... I been able to make it sound normal in a fresh project and it seems something related to pulse, alsa or eveything... I think its some kind of absurd autogain

**Comment by doradoro** [comment]
I reinstalled pulse and removed everythign related to sound. I reinstalled flowblade and only needed nox as a dependency. I can hear the sound perfectly in a new project but, in the old one, keeps making weird things. So, it seems some kind of configuration thing...

**Comment by doradoro** [comment]
I deleted all caches and now xml files are missing so i cant open the flb file again becasue i cant relink nothing ¬_¬

**Comment by doradoro** [comment]
SOLVED! There was some kind of cache or something changing the file volumes!

I solved the compound clips missing xmls attaching an empty file empty.xml to them

**Comment by doradoro** [comment]
Broken again! using xml compound clips in the project lower the volume from the entire project! O_o

**Comment by doradoro** [comment]
Problem detected! each compound created clip decrease the volume 25%

**Comment by jliljebl** [comment]
Ok, let's try to reproduce this.

**Comment by jliljebl** [comment]
No, compound clips do not create audio level changes here. I did find a Python 3 port bug but audio was not affected by compound clips.

**Comment by doradoro** [comment]
If i create a new sequence in the same project, the sound bars doesnt appears but if i play the file in the media tab or timeline it sounds ok

**Comment by jliljebl** [comment]
Ok, will testi later.

to 31. lokakuuta 2019 klo 23.07 doradoro <notifications@github.com>
kirjoitti:

**Comment by doradoro** [comment]
As you can see the volume changes for no visible reason

https://youtu.be/XqMQRa_vD3w

**Comment by jliljebl** [comment]
"This video is private."

**Comment by doradoro** [comment]
solved, sorry i got confused with "hidden" and "private" meanings

**Comment by jliljebl** [comment]
Ok, I got it. This one is real.

**Comment by jliljebl** [comment]
It is just that the monitor widget dies on other sequences, playback and rendering not affected. I think I only start looking at audio when I render output which happens to be on first sequence and use other sequences for silent video stuff mostly.

But anyway good input, this needs fixing.

**Comment by doradoro** [comment]
mmm i will try to copy and paste the entire sequence to another one and see what happens

**Comment by jliljebl** [comment]
I fixed "the monitor widget dead on additional sequences" bug and pushed a patch to repo. Going to bed and will probably be off project for few days but if you additional info please post here, I'll look at it when I do next session.

**Comment by doradoro** [comment]
i think i just know what it is. i dont have swap configured so my system kills everything that excedes my 8gb of ram. it seems taht having a compund clip inside another compound clip is too much for my ram and then weird things start to happen without any error shown...

https://youtu.be/-7tHbpA4aR4

**Comment by dvdlvr** [comment]
If it's a RAM issue you could try rendering with the melt command line (After closing Flowblade and any other open apps).
You can do this as follows:
In Flowblade, finish editing your project, then export the MLT XML (File -> Export).
Go to the 'Render' tab, set your render settings and then click on 'Render with arguments'. Copy all the arguments that appear in the box.
On the command line type the following:
melt name_of_xml.xml -consumer avformat:"outfile_name.mkv [copy of all the args here]"
That should work. The copy of all the args is in the format keyword=value keyword=value (separated by spaces) as displayed in Flowblade.

**Comment by doradoro** [comment]
I turned on swap and opened the project with all the other things closed in a fresh restart but the sound keep changing itself...

im going to try to render it and see what happens

**Comment by doradoro** [comment]
The sound is lowered in a plain new project but seems that .xml files increase that lowering 
effect much more...

![Captura de pantalla -2019-10-31 23-53-35](https://user-images.githubusercontent.com/495093/67991486-b1553e80-fc39-11e9-8b6c-a14c82b60e8b.png)

**Comment by doradoro** [comment]
To obtain similar volume levels i need to add an audio effect of 150% in volume increase. That effect happens in a second sequence too. I dont understand whats is the relationship with compound clips.

**Comment by doradoro** [comment]
Using:

melt prueba.xml -consumer avformat:"outfile_name.mkv" f=mkv acodec=flac vcodec=ffv1 level=3 g=1 slicecrc=1 slices=16

The result is the same

**Comment by doradoro** [comment]
Using "File Luma To Alpha" Filter is what decreases the volume of the entire sequence!

Thresold value affects total volume!

I thought that was the compound clips but i used mkv and mp4 files with the same result...

https://youtu.be/nnBk5k3efpc

**Comment by jliljebl** [comment]
Ok, thanks, will investigate next week.

pe 1. marraskuuta 2019 klo 14.18 doradoro <notifications@github.com>
kirjoitti:

**Comment by doradoro** [comment]
As you can see, all the waveform's timelines are about half high they could be. Something is changing the volumes from the very first moment. Hope i could help more!

**Comment by jliljebl** [comment]
Great,  confirmed and fixed. MLT filter "shape" had undocumented param "audio_match" that was needed to set to 0 to not affect audio, on docs we got zip, I'll file bug report later.

https://github.com/mltframework/mlt/blob/master/src/modules/vmfx/filter_shape.yml

I really did almost all by testing with this on mute images and when I used video clips I must have just looked at what is happening on image and mentally ignored all audio. A bit of an example of not been able to perceive something that is right if front of you if you are not mentally predisposed to making the observation.

Thanks for your persistence on this, we got couple of big bugs fixed. Flowblade already has a lot of features and my usual test edits focus on currently on going development features and stuff just gets missed.so user feedback is essential for improving quality.
